File: problems/3349-adjacent-increasing-subarrays-detection-i/solution.java
import java.util.List;

class Solution {
    public boolean hasIncreasingSubarrays(List<Integer> nums, int k) {
        for (int a = 0; a <= nums.size() - 2 * k; a++) {
            int b = a + k;
            boolean increasingA = true;
            for (int i = a; i < a + k - 1; i++) {
                if (nums.get(i) >= nums.get(i + 1)) {
                    increasingA = false;
                    break;
                }
            }
            boolean increasingB = true;
            for (int i = b; i < b + k - 1; i++) {
                if (nums.get(i) >= nums.get(i + 1)) {
                    increasingB = false;
                    break;
                }
            }
            if (increasingA && increasingB) {
                return true;
            }
        }
        return false;
    }
}

File: problems/3349-adjacent-increasing-subarrays-detection-i/solution.cpp
class Solution {
public:
    bool hasIncreasingSubarrays(vector<int>& nums, int k) {
        for (int a = 0; a <= nums.size() - 2 * k; ++a) {
            int b = a + k;
            bool increasingA = true;
            for (int i = a; i < a + k - 1; ++i) {
                if (nums[i] >= nums[i + 1]) {
                    increasingA = false;
                    break;
                }
            }
            if (!increasingA) continue;

            bool increasingB = true;
            for (int i = b; i < b + k - 1; ++i) {
                if (nums[i] >= nums[i + 1]) {
                    increasingB = false;
                    break;
                }
            }
            if (increasingB) return true;
        }
        return false;
    }
};